internal/handler/post_task_execution_request_handler.go

```diff
@@ -60,7 +60,24 @@ func NewPostTaskExecutionRequestHandler(config Config) func(w http.ResponseWrite
 
 		err = db.Put([]byte(enum.LastTaskKey), []byte(providerResponse.String()), nil)
 		if err != nil {
-			log.Println("failed to save the task")
+			log.Println("failed to save the last task")
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		task, err := json.Marshal(&entity.TaskStatistic{
```

Comment:
`json.Marshal` does not really require the "address of" operator (`&`) , so why no to remove it?